src/ui/SmartSnippet/UserFeedbackBanner.ts

```diff
@@ -0,0 +1,170 @@
+import { $$ } from '../../utils/Dom';
+import { l } from '../../strings/Strings';
+import { AccessibleButton } from '../../utils/AccessibleButton';
+import { SVGIcons } from '../../utils/SVGIcons';
+
+const ROOT_CLASSNAME = 'coveo-user-feedback-banner';
+const CONTAINER_CLASSNAME = `${ROOT_CLASSNAME}-container`;
+const LABEL_CLASSNAME = `${ROOT_CLASSNAME}-label`;
+const BUTTONS_CONTAINER_CLASSNAME = `${ROOT_CLASSNAME}-buttons`;
+const YES_BUTTON_CLASSNAME = `${ROOT_CLASSNAME}-yes-button`;
+const NO_BUTTON_CLASSNAME = `${ROOT_CLASSNAME}-no-button`;
+const BUTTON_ACTIVE_CLASSNAME = `${ROOT_CLASSNAME}-button-active`;
+const THANK_YOU_BANNER_CLASSNAME = `${ROOT_CLASSNAME}-thanks`;
+const THANK_YOU_BANNER_ACTIVE_CLASSNAME = `${THANK_YOU_BANNER_CLASSNAME}-active`;
+const ICON_CLASSNAME = `${THANK_YOU_BANNER_CLASSNAME}-icon`;
+const EXPLAIN_WHY_CLASSNAME = `${ROOT_CLASSNAME}-explain-why`;
+const EXPLAIN_WHY_ACTIVE_CLASSNAME = `${EXPLAIN_WHY_CLASSNAME}-active`;
+
+enum UsefulState {
+  Unknown,
+  Yes,
+  No
+}
+
+interface IButtonOptions {
+  text: string;
+  className: string;
+  action: Function;
+  icon?: {
+    content: string;
+    className: string;
+  };
+}
+
+export const UserFeedbackBannerClassNames = {
+  ROOT_CLASSNAME,
+  CONTAINER_CLASSNAME,
+  LABEL_CLASSNAME,
+  BUTTONS_CONTAINER_CLASSNAME,
+  YES_BUTTON_CLASSNAME,
+  NO_BUTTON_CLASSNAME,
+  BUTTON_ACTIVE_CLASSNAME,
+  THANK_YOU_BANNER_CLASSNAME,
+  THANK_YOU_BANNER_ACTIVE_CLASSNAME,
+  ICON_CLASSNAME,
+  EXPLAIN_WHY_CLASSNAME,
+  EXPLAIN_WHY_ACTIVE_CLASSNAME
+};
+
+export enum UserFeedbackExplaination {
+  NotUseful,
+  Incorrect,
+  HardToRead
+}
+
+export class UserFeedbackBanner {
+  private isUseful = UsefulState.Unknown;
+  private yesButton: HTMLElement;
+  private noButton: HTMLElement;
+  private thankYouBanner: HTMLElement;
+  private explainWhy: HTMLElement;
+
+  constructor(
+    private readonly sendUsefulnessAnalytics: (isUseful: boolean) => void,
+    private readonly sendExplainationAnalytics: (explaination: UserFeedbackExplaination) => void
+  ) {}
+
+  public build() {
+    return $$(
+      'div',
+      {
+        className: ROOT_CLASSNAME
+      },
+      this.buildContainer(),
+      this.buildThankYouBanner()
+    ).el;
+  }
+
+  private buildContainer() {
+    return $$(
+      'div',
+      {
+        className: CONTAINER_CLASSNAME
+      },
+      this.buildLabel(),
+      this.buildButtons()
+    ).el;
+  }
+
+  private buildLabel() {
+    return $$('span', { className: LABEL_CLASSNAME }, l('UsefulnessFeedbackRequest')).el;
+  }
+
+  private buildThankYouBanner() {
+    return (this.thankYouBanner = $$(
```

Comment:
I find it a bit hard to read when we imbrick a ton of element inside one another like that with `$$` .

I'd prefer if they were created one by one, then appended, just a bit easier to read.

Same for the `buildButtons` method

Reply:
Alright, I'll use appends instead.